### ISSUE TITLE
[Android] Datepicker closes only the second time

### ISSUE THREAD

**Issue by fedorov-xyz** [henninghall/react-native-date-picker#415]
`react-native-date-picker` version is `4.1.1`.

**Describe the bug**
After selecting a date and clicking on "Confirm" the modal window closes and opens again. You must click "Confirm" again to close it permanently. 

In this video I open datepicker and then click "Confirm":

<img src="https://user-images.githubusercontent.com/25563268/139584358-de21b892-dacf-453d-989e-833d07611c6a.gif" alt="" width="250"/>

**To Reproduce**
```tsx
export const App = () => {
  const [showDatePicker, setShowDatePicker] = useState(false);
  const [birthdate, setBirthdate] = useState(new Date());

  return (
    <>
      <Button title="Open" onPress={() => setShowDatePicker(true)} />

      <DatePicker
        modal
        open={showDatePicker}
        date={birthdate}
        mode="date"
        onConfirm={(date) => {
          setBirthdate(date);
          setShowDatePicker(false);
        }}
        onCancel={() => {
          setShowDatePicker(false);
        }}
      />
    </>
  );
};
```

I thinks this is happening because two `setState` are called in one time inside `onConfirm` handler. This bug is solved with wrapping `setState` calls in `unstable_batchedUpdates`:

```tsx
import { unstable_batchedUpdates } from 'react-native';

onConfirm={(date) => {
  unstable_batchedUpdates(() => {
    setBirthdate(date);
    setShowDatePicker(false);
  });
}}
```

**Smartphone (please complete the following information):**
 - OS: Android
 - `react-native info`:

```
  System:
    OS: macOS 12.0.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 36.68 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.17.0 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 6.14.13 - /usr/local/bin/npm
    Watchman: 2021.09.13.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.2 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.0.1, iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0
    Android SDK: Not Found
  IDEs:
    Android Studio: 2020.3 AI-203.7717.56.2031.7678000
    Xcode: 13.1/13A1030d - /usr/bin/xcodebuild
  Languages:
    Java: javac 17 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.66.0 => 0.66.0 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

**Comment by henninghall** [comment]
Hi, thanks for reporting this. I am aware of this issue that happens when the set function is called before the show(false) function. Try change the order of the two and let me know if it solved it for you!

**Comment by fedorov-xyz** [comment]
Changing the order of `setState` calls does solve the problem.  But I think this is an unobvious point. What do you think about explicitly talking about it in the Readme?

**Comment by henninghall** [comment]
I definitely agree! Also, the order shouldn't really matter so I'll keep this issue open until it's fixed.

**Comment by prox2** [comment]
> Hi, thanks for reporting this. I am aware of this issue that happens when the set function is called before the show(false) function. Try change the order of the two and let me know if it solved it for you!

this doesn't solve it for me, any update on this case?

**Comment by daominhsangvn** [comment]
The order of the set state actually matters. It solved my case

**Comment by biskis** [comment]
For most cases the order fix the problem. But not all the time. Sometimes randomly it still shows multiple times on the screen.

**Comment by zahidahmed024** [comment]
i also faced this problem, in my case , i had `useEffect( () => {...api call...}, [date1,date2] ) . `  the issue solved when i removed date1,date2 from third parameter

**Comment by henninghall** [comment]
A fix for this released in v4.1.4
